lib/system/freertos/mutex.h

```diff
@@ -43,39 +42,39 @@ typedef struct {
 static inline void __metal_mutex_init(metal_mutex_t *mutex)
 {
 	metal_assert(mutex);
-	mutex->m = xSemaphoreCreateMutex();
-	metal_assert(mutex->m != NULL);
+	atomic_store(&mutex->v, 0);
+	metal_assert(mutex->v);
```

Comment:
metal_assert(mutex->v) is not necessary, and the expect value is 0 but not non-zero